src/lib/snack-bar/snack-bar.ts

```diff
@@ -226,32 +226,24 @@ export class MatSnackBar {
    * @param config The user-specified snack bar config.
    */
   private _createOverlay(config: MatSnackBarConfig): OverlayRef {
-    const overlayConfig = new OverlayConfig();
-    overlayConfig.direction = config.direction;
+    const {horizontalPosition, verticalPosition, direction} = config;
+    const positionStrategy = this._overlay.position().global();
+    const overlayConfig = new OverlayConfig({direction, positionStrategy});
 
-    let positionStrategy = this._overlay.position().global();
     // Set horizontal position.
-    const isRtl = config.direction === 'rtl';
-    const isLeft = (
-      config.horizontalPosition === 'left' ||
-      (config.horizontalPosition === 'start' && !isRtl) ||
-      (config.horizontalPosition === 'end' && isRtl));
-    const isRight = !isLeft && config.horizontalPosition !== 'center';
-    if (isLeft) {
-      positionStrategy.left('0');
-    } else if (isRight) {
-      positionStrategy.right('0');
-    } else {
+    if (!horizontalPosition || horizontalPosition === 'center') {
```

Comment:
Since it defaults to centered horizontally?  We should only have to call for a horizontal position if it's not center correct?

```ts
if (horizontalPosition && horizontalPosition !== 'center') {
  positionStrategy[horizontalPosition]('0');
}
```